tracing-core/src/metadata.rs

```diff
@@ -309,28 +332,209 @@ impl FromStr for Level {
+    ///
+    /// Designates hazardous situations.
+    Warn = 3,
+    /// The "error" level.
+    ///
+    /// Designates very serious errors.
+    Error = 4,
+}
+
+// === impl LevelFilter ===
+
+impl From<Level> for LevelFilter {
+    #[inline]
+    fn from(level: Level) -> Self {
+        Self::from_level(level)
+    }
+}
+
+impl Into<Option<Level>> for LevelFilter {
+    #[inline]
+    fn into(self) -> Option<Level> {
+        self.into_level()
+    }
+}
+
+impl LevelFilter {
+    /// The "off" level.
+    ///
+    /// Designates that trace instrumentation should be completely disabled.
+    pub const OFF: LevelFilter = LevelFilter(None);
     /// The "error" level.
     ///
     /// Designates very serious errors.
-    Error = 1,
+    pub const ERROR: LevelFilter = LevelFilter::from_level(Level::ERROR);
     /// The "warn" level.
     ///
     /// Designates hazardous situations.
-    Warn,
+    pub const WARN: LevelFilter = LevelFilter::from_level(Level::WARN);
     /// The "info" level.
     ///
     /// Designates useful information.
-    Info,
+    pub const INFO: LevelFilter = LevelFilter::from_level(Level::INFO);
     /// The "debug" level.
     ///
     /// Designates lower priority information.
-    Debug,
+    pub const DEBUG: LevelFilter = LevelFilter::from_level(Level::DEBUG);
     /// The "trace" level.
     ///
     /// Designates very low priority, often extremely verbose, information.
-    Trace,
+    pub const TRACE: LevelFilter = LevelFilter(Some(Level::TRACE));
+
+    /// Returns a `LevelFilter` that enables spans and events with verbosity up
+    /// to and including `level`.
+    pub const fn from_level(level: Level) -> Self {
+        Self(Some(level))
+    }
+
+    /// Returns the most verbose [`Level`] that this filter accepts, or `None`
+    /// if it is [`OFF`].
+    ///
+    /// [`Level`]: ../struct.Level.html
+    /// [`OFF`]: #associatedconstant.OFF
+    pub const fn into_level(self) -> Option<Level> {
+        self.0
+    }
+
+    // These consts are necessary because `as` casts are not allowed as
+    // match patterns.
+    const ERROR_USIZE: usize = LevelInner::Error as usize;
+    const WARN_USIZE: usize = LevelInner::Warn as usize;
+    const INFO_USIZE: usize = LevelInner::Info as usize;
+    const DEBUG_USIZE: usize = LevelInner::Debug as usize;
+    const TRACE_USIZE: usize = LevelInner::Trace as usize;
+    // Using the value of the last variant + 1 ensures that we match the value
+    // for `Option::None` as selected by the niche optimization for
+    // `LevelFilter`. If this is the case, converting a `usize` value into a
+    // `LevelFilter` (in `LevelFilter::max`) will be an identity conversion,
+    // rather than generating a lookup table.
+    const OFF_USIZE: usize = LevelInner::Error as usize + 1;
+
+    /// Returns a `LevelFilter` that matches the most verbose [`Level`] that any
+    /// currently active [`Subscriber`] will enable.
+    #[inline(always)]
+    pub fn max() -> Self {
```

Comment:
Interesting fact: `log` has [a `LevelFilter::max` with the same signature](https://github.com/rust-lang/log/blob/ca54ac75ce2a9f963cd366f0d68e40e667e03ff4/src/lib.rs#L651-L655) that returns a totally different thing. Maybe we should rename this to avoid confusing users familiar with the `log` API...

I am not really sure if there's all that much value in having a method for returning the highest _possible_ level filter, when users can just write `TRACE`...but maybe people are doing that. Also, I would have probably called that function `all()`, but, whatever...

Reply:
@davidbarsky @LucioFranco any thoughts on the API naming before I merge this PR?